<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>570</width>
    <height>752</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Crypto</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normalon>crypto.png</normalon>
   </iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_23">
    <item row="0" column="0">
     <widget class="QTabWidget" name="maintab">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="cryption">
       <attribute name="title">
        <string>加密/解密</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="2" column="0">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QLabel" name="cry_key_in_label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>密钥输入</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="cry_key_in"/>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="cry_key_in_open">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <widget class="QGroupBox" name="cry_file">
            <property name="title">
             <string>文件</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0" colspan="3">
              <widget class="QTreeWidget" name="cry_file_list">
               <column>
                <property name="text">
                 <string>文件</string>
                </property>
               </column>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="cry_file_remove">
               <property name="text">
                <string>删除</string>
               </property>
               <property name="flat">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="1" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <property name="spacing">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="cry_file_add">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>75</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPushButton {
    margin-right: -18px;
}</string>
                 </property>
                 <property name="text">
                  <string>添加...</string>
                 </property>
                 <property name="checkable">
                  <bool>false</bool>
                 </property>
                 <property name="default">
                  <bool>false</bool>
                 </property>
                 <property name="flat">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="cry_file_add_menu">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>18</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPushButton {
    margin-left: -8px;
}</string>
                 </property>
                 <property name="flat">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="2">
             <widget class="QComboBox" name="cry_algorithm">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <item>
               <property name="text">
                <string>AES</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>DES</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>RSA-1024</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>RSA-2048</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>RSA-4096</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-secp256k1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-secp256r1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-SM2</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>算法选择</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="cry_begin">
              <property name="text">
               <string>开始</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QGroupBox" name="cry_operation">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>操作选择</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QRadioButton" name="cry_operation_enc">
               <property name="text">
                <string>加密</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="cry_operation_dec">
               <property name="text">
                <string>解密</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gen">
       <attribute name="title">
        <string>密钥对生成</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_12">
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>小提示：只有非对称加密算法才需要生成密钥对。</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="2">
             <widget class="QComboBox" name="gen_algorithm">
              <item>
               <property name="text">
                <string>RSA-1024</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>RSA-2048</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>RSA-4096</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-secp256k1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-secp256r1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ECC-SM2</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>算法选择</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gridLayout_9">
            <item row="0" column="1">
             <widget class="QLineEdit" name="gen_pri"/>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="gen_pub"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>公钥保存路径</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="gen_pub_open">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>私钥保存路径</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="gen_pri_open">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QGridLayout" name="gridLayout_11">
            <item row="0" column="0">
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="pushButton">
              <property name="text">
               <string>生成</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <spacer name="horizontalSpacer_5">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="hash">
       <attribute name="title">
        <string>哈希</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_22">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_15">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_13">
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>算法选择</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="hash_algorithm">
                <item>
                 <property name="text">
                  <string>MD5</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>SHA256</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QGroupBox" name="hash_operation">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>操作选择</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_18">
               <item row="1" column="0">
                <widget class="QRadioButton" name="hash_operation_file">
                 <property name="text">
                  <string>文件操作</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QRadioButton" name="hash_operation_string">
                 <property name="text">
                  <string>字符串操作</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QGroupBox" name="hash_file">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>文件输入</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_17">
               <item row="0" column="0">
                <layout class="QGridLayout" name="gridLayout_16">
                 <item row="0" column="1">
                  <widget class="QPushButton" name="hash_file_in_open">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>30</width>
                     <height>16777215</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>...</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="0">
                  <widget class="QLineEdit" name="hash_file_in"/>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QGroupBox" name="hash_string">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>字符串输入</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_19">
               <item row="0" column="0">
                <widget class="QTextEdit" name="hash_string_in">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QGridLayout" name="gridLayout_21">
            <item row="0" column="0">
             <widget class="QGroupBox" name="hash_result">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>结果</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_14">
               <item row="0" column="0">
                <widget class="QLineEdit" name="hash_result_out"/>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <layout class="QGridLayout" name="gridLayout_20">
              <item row="0" column="0">
               <spacer name="horizontalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="hash_begin">
                <property name="text">
                 <string>开始</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <spacer name="horizontalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <spacer name="verticalSpacer_for_hash_file">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>maintab</tabstop>
  <tabstop>cry_algorithm</tabstop>
  <tabstop>cry_operation_enc</tabstop>
  <tabstop>cry_operation_dec</tabstop>
  <tabstop>cry_key_in</tabstop>
  <tabstop>cry_key_in_open</tabstop>
  <tabstop>cry_file_list</tabstop>
  <tabstop>cry_file_add</tabstop>
  <tabstop>cry_file_add_menu</tabstop>
  <tabstop>cry_file_remove</tabstop>
  <tabstop>cry_begin</tabstop>
  <tabstop>gen_algorithm</tabstop>
  <tabstop>gen_pri</tabstop>
  <tabstop>gen_pri_open</tabstop>
  <tabstop>gen_pub</tabstop>
  <tabstop>gen_pub_open</tabstop>
  <tabstop>hash_algorithm</tabstop>
  <tabstop>hash_operation_file</tabstop>
  <tabstop>hash_operation_string</tabstop>
  <tabstop>hash_string_in</tabstop>
  <tabstop>hash_file_in</tabstop>
  <tabstop>hash_file_in_open</tabstop>
  <tabstop>hash_result_out</tabstop>
  <tabstop>hash_begin</tabstop>
  <tabstop>pushButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
